packages/babel-parser/src/parser/statement.js

```diff
@@ -2255,32 +2255,23 @@ export default class StatementParser extends ExpressionParser {
 
       // parse AssertionKey : IdentifierName, StringLiteral
       const keyName = this.state.value;
-      if (this.match(tt.string)) {
-        node.key = this.parseStringLiteral(keyName);
-      } else {
-        node.key = this.parseIdentifier(true);
-      }
-      this.expect(tt.colon);
-
-      // for now we are only allowing `type` as the only allowed module attribute
-      if (keyName !== "type") {
-        this.raise(
-          node.key.start,
-          Errors.ModuleAttributeDifferentFromType,
```

Comment:
Note that we can not remove ModuleAttributeDifferentFromType from Error templates because it is still used in `parseModuleAttributes`.